samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+1. Azure Service Fabric [dev cluster](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-get-started) running and configured with 5 nodes.
+1. An Azure Service Bus namespace that can be used for communication between the instances.
+1. A system environment variable named "AzureServiceBus.ConnectionString" set to the connection string of the Azure Service Bus namespace. The connection string must provide Manage rights.
+
+Note: A Service Fabric cluster runs under the Network Service account and only reads system environment variables. Make sure the environment variable "AzureServiceBus.ConnectionString" is defined as a system environment variable and not user-scoped.
+
+
+## Scenario
+
+The scenario used in this sample covers a voting system. In this voting system the cast votes are counted by candidate. The cast votes are `published` from the voter client to the endpoint responsible for counting candidate votes.
+
+Next to this the system also counts the total number of votes cast in each zip code. In order to achieve this the candidate voting endpoint issues a `request` to the zip code counting endpoint to track the zip code. The zip code counting endpoint will `reply` back with the intermediary results.
+
+When the election is closed, the candidate vote counting endpoint will `publish` the results per candidate and report them using Service Fabric diagnostics infrastructure (ETW Event Viewer windows).
+
+After the counting time expires, using a `timeout`, the zip code counting endpoint `sends a local command` to report the statistics per zip code.
+
+For sake of simplicity, there are only 2 candidates in the election, called "John" and "Abby". Zip codes are always assumed to be valid integers with a length up to 5 digits in the range of 0 to 99000.
+
+```mermaid
+sequenceDiagram
+
+participant v as Voter
+participant cv as CandidateVoteCounts
+participant zc as ZipCodeCounts
+participant cvt as 1min Timeout
+participant zct as 1min Timeout
+
+v->> cv: 1. Publish VotePlaced
+
+cv ->> zc: 2. Send
+zc -->> cv: 3. Reply
+
+v->> cv: 4. Send ElectionClosed
+
+cv ->>+ cvt: 5. Request Timeout
+cvt-->>- cv: 6. Reply
+cv ->> cv: 7. Publish
+
+zc ->>+ zct: 8. Request Timeout
+zct -->>- zc: 9. Reply
+zc ->> zc: 10. Publish
+```
+
+### Trade offs and known limitations
+
+The scenario has been set up to show the different kinds of communication patterns that can occur in a partitioned solution: `send`, `send local`, `publish/subscribe`, `request/reply`, `timeout`.
+
+The downside of the focus on the communication patterns is that the saga design is less than ideal for a real voting system. There will be quite some contention on the saga data, which may result in concurrency exceptions and a few retries impacting performance of the system.
+
+For logging purposes a simple static logger was used to expose specific log statements for the routing part. It is recommended to use a dedicated logging package for production instead.
```

Comment:
does SF work with any logging package? or only ETW? is there a link for "SF logging" we can use?